code/backend/pages/01_Ingest_Data.py

```diff
@@ -75,21 +77,32 @@ def remote_convert_files_and_add_embeddings(process_all=False):
 
 
 def add_urls():
-    params = {}
-    if env_helper.FUNCTION_KEY is not None:
-        params["code"] = env_helper.FUNCTION_KEY
-        params["clientId"] = "clientKey"
     urls = st.session_state["urls"].split("\n")
-    for url in urls:
-        body = {"url": url}
-        backend_url = urllib.parse.urljoin(
-            env_helper.BACKEND_URL, "/api/AddURLEmbeddings"
-        )
-        r = requests.post(url=backend_url, params=params, json=body)
-        if not r.ok:
-            raise ValueError(f"Error {r.status_code}: {r.text}")
-        else:
-            st.success(f"Embeddings added successfully for {url}")
+    if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
+        try:
```

Comment:
It might be worth pulling out the contents of the if/else into separate funtions, one for the download + upload approach and one for the existing add url embeddings approach.